### PR TITLE
feat: implement CICDv2 (Java&Go)

### PR DESCRIPTION
**Pull request by panpuchkov**

### CODE REVIEW

File: .github/workflows/BuildPushGO.yaml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        default: 'cariq'
+        type: string
+      image-tag:
+        default: '0.1.0'
+        type: string
+      sonar:
+        default: 'https://sonar.gocariq.com'
+        type: string
+    secrets:
+      SLACK_WEBHOOK_URL:
```
Comment (alexgocariq) [inline .github/workflows/BuildPushGO.yaml:29]: not sue if slack webhook should be required = true

File: .github/workflows/BuildPushJava.yaml [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        id: validate-chart
+        run: |
+          helm lint ./charts/${{ inputs.app-name }}/
+
+      - name: Set up JDK 11
+        if: ${{ env.IMAGE_EXIST == 'false' }}
+        id: setting-env
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
```
Comment (alexgocariq) [inline .github/workflows/BuildPushJava.yaml:99]: we can user version as env variable I think we can also reference it in the step name so we do not need to use hard coded value in the step name

File: .github/workflows/BuildPushJava.yaml [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      GKE_SA_GHA_PUSHER_CREDENTIALS:
+        required: true
+      GHA_DEVOPS_DISPATCH:
+        required: true
+      K8S_POD_DEPLOY_VALIDATION:
+        required: true
+      GHA_DISPATCH_TEST:
+        required: true
+jobs:
+  build:
```
[the remark this answers is not in the document]
Comment (Mo6niy) [reply .github/workflows/BuildPushJava.yaml:46]: WAs added condition in reusable workflow for Main flow
      - name: Check if branch is not main
        if: ${{ github.ref != 'refs/heads/main' }}
        run: |
          echo "This branch is not main."
          echo "Current branch is ${{ github.ref_name }}"
          exit1

File: .github/workflows/BuildPushJava.yaml [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+      - name: Dockerfile lint
+        if: ${{ env.IMAGE_EXIST == 'false' }}
+        id: validate-dockerfile
+        shell: bash
+        run: |
+          docker run --rm -e HADOLINT_IGNORE=DL3007,DL3008,DL3018 -i hadolint/hadolint < Dockerfile
+
+      - name: helm lint
+        if: ${{ env.IMAGE_EXIST == 'false' }}
+        id: validate-chart
```
[the remark this answers is not in the document]
Comment (Mo6niy) [reply .github/workflows/BuildPushJava.yaml:90]: Done

File: .github/workflows/BuildPushJava.yaml (hunk 2 above)
Comment (Mo6niy) [reply .github/workflows/BuildPushJava.yaml:99]: This step is going to be removed in future

File: .github/workflows/BuildPushJava.yaml [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+      #     targetEnv: ${{ inputs.deploy-env }}
+      #     tag: ${{ env.SHORT_SHA }}
+      #     workFlowFileName: core-service-test
+
+      - uses: act10ns/slack@v2.0.0
+        with:
+          status: ${{ job.status }}
+          steps: ${{ toJson(steps) }}
+          channel: ${{ inputs.slack-channel }}
+        if: always()
```
[the remark this answers is not in the document]
Comment (Mo6niy) [reply .github/workflows/BuildPushJava.yaml:181]: has to be there for situation when we got fail

File: .github/workflows/BuildPushMain.yaml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        default: 'cariq'
+        type: string
+
+    secrets:
+      GOOGLE_CRED:
+        required: true
+      K8S_POD_DEPLOY_VALIDATION:
+        required: true
+jobs:
+  build:
```
[the remark this answers is not in the document]
Comment (Mo6niy) [reply .github/workflows/BuildPushMain.yaml:23]: done

File: .github/workflows/BuildPushMain.yaml [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+      - name: Image existence check
+        id: check-image
+        uses: gocariq-gitops/release-actions/checkDockerImage@reusable_workflows
+        with:
+          google_credentials: ${{secrets.GOOGLE_CRED}}
+          google_project_id: ${{ inputs.project-id }}
+          docker_repo_name:  ${{ inputs.registry-name }}
+          docker_image_name: ${{ inputs.app-name }}
+
+      - name: test echo
```
[the remark this answers is not in the document]
Comment (Mo6niy) [reply .github/workflows/BuildPushMain.yaml:54]: done

File: .github/workflows/BuildPushMain.yaml [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        run: |
+          echo "IMAGE_EXIST - ${{ env.IMAGE_EXIST }}"
+          echo "SHORT_SHA   -${{ env.SHORT_SHA }}"
+
+      - name: STOP if Image isn't exist
+        if: ${{ env.IMAGE_EXIST == 'false' }}
+        run: |
+            echo "Image is not exist in Dev Repo. Stopping Pipeline.."
```
[the remark this answers is not in the document]
Comment (Mo6niy) [reply .github/workflows/BuildPushMain.yaml:62]: done

File: .github/workflows/BuildPushMain.yaml [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+            exit 1
+
+      # - name: Validate INT environment
```
[the remark this answers is not in the document]
Comment (Mo6niy) [reply .github/workflows/BuildPushMain.yaml:65]: Done

File: .github/workflows/BuildPushMain.yaml [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+      #   with:
+      #     google_credentials: ${{ secrets.K8S_POD_DEPLOY_VALIDATION }}
+      #     app_name: ${{ inputs.app-name }}
+      #     google_project_id: ${{ inputs.project-id }}
+      #     expected_git_tag: ${{ env.SHORT_SHA }}
+      #     env_name: int
+      #     max_attempt: 2
+      #     sleep_time: 10
+
+      # - name: Validate STAGE environment
```
[the remark this answers is not in the document]
Comment (Mo6niy) [reply .github/workflows/BuildPushMain.yaml:76]: Done

File: .github/workflows/BuildPushMain.yaml [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+      - name: Docker Auth
+        shell: bash
+        run: |
+          gcloud auth configure-docker us-docker.pkg.dev
+
+      - name: Get Next Version
+        shell: bash
+        run: |
+          # docker run --rm -v $(pwd):/app -w /app --entrypoint "" us-docker.pkg.dev/gocariq-platform/cariq/semver-helper \
```
[the remark this answers is not in the document]
Comment (Mo6niy) [reply .github/workflows/BuildPushMain.yaml:102]: done

File: .github/workflows/BuildPushMain.yaml [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+          # sh -c "git config --global --add safe.directory /app && semver-helper --next-ver"
+          echo "NEW_VERSION=$(docker run --rm -v $(pwd):/app -w /app --entrypoint "" us-docker.pkg.dev/gocariq-platform/cariq/semver-helper \
+          sh -c "git config --global --add safe.directory /app && semver-helper --next-ver")" >> $GITHUB_ENV
+
+      - name: Create PROD version of Image
+        run: |
+          echo "New PROD version of image - ${{ env.NEW_VERSION }}"
```
[the remark this answers is not in the document]
Comment (Mo6niy) [reply .github/workflows/BuildPushMain.yaml:109]: done

**Pull request closed without merging**